cpp/src/arrow/visitor_generate.h

```diff
@@ -65,4 +65,50 @@ namespace arrow {
   ACTION(Dictionary);                           \
   ACTION(Extension)
 
+#define ARROW_GENERATE_FOR_ALL_INTEGER_TYPES2(ACTION, ARG1) \
+  ACTION(Int8, ARG1);                                       \
+  ACTION(UInt8, ARG1);                                      \
+  ACTION(Int16, ARG1);                                      \
+  ACTION(UInt16, ARG1);                                     \
+  ACTION(Int32, ARG1);                                      \
+  ACTION(UInt32, ARG1);                                     \
+  ACTION(Int64, ARG1);                                      \
+  ACTION(UInt64, ARG1)
+
+#define ARROW_GENERATE_FOR_ALL_NUMERIC_TYPES2(ACTION, ARG1) \
+  ARROW_GENERATE_FOR_ALL_INTEGER_TYPES2(ACTION, ARG1);      \
+  ACTION(HalfFloat, ARG1);                                  \
+  ACTION(Float, ARG1);                                      \
+  ACTION(Double, ARG1)
+
+#define ARROW_GENERATE_FOR_ALL_TYPES2(ACTION, ARG1)    \
```

Comment:
There were some changes in this file since I did this and there's a better approach I can take that doesn't need these new lines here.